### PR TITLE
adapter: First step in renaming mz_introspection role to mz_support

### PR DESCRIPTION
**Pull request by andrewrodriguez-m**
I'm renaming the system role id that used to be called mz_introspection to mz_support. This requires special logic in authentication to use the mz_support role when the mz_introspection is provided but this logic can be removed as soon as no one uses mz_introspection.

Another approach I tried was to add a new system role, grant it the same access that mz_introspection has and then eventually delete the mz_introspection role but couldn't think of a way to avoid leaving a gap in the system ids assigned.

Progress on #20553

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by benesch** [comment]
s/mz_system/mz_support/ in the PR description, I think!

File: src/sql/src/session/user.rs [hunk 1]
```diff
@@ -17,18 +17,36 @@ pub static SYSTEM_USER: Lazy<User> = Lazy::new(|| User {
...
 });
 
 pub static INTERNAL_USER_NAMES: Lazy<BTreeSet<String>> = Lazy::new(|| {
-    [&SYSTEM_USER, &INTROSPECTION_USER]
+    let mut names: BTreeSet<String> = [&SYSTEM_USER, &SUPPORT_USER]
         .into_iter()
         .map(|user| user.name.clone())
-        .collect()
+        .collect();
+    names.insert("mz_introspection".to_string()); // TODO: remove
```
Comment (mjibson) [inline src/sql/src/session/user.rs:35]:
```suggestion
    names.insert(INTROSPECTION_USER.name.clone()); // TODO: remove
```

File: test/legacy-upgrade/check-from-v0.53.0-role.td [hunk 2]
```diff
@@ -14,7 +14,7 @@ SELECT mz_version_num() < 5200;
 group
 joe
 mz_system
-mz_introspection
+mz_support
```
[the remark this answers is not in the document]
Comment (andrewrodriguez-m) [reply test/legacy-upgrade/check-from-v0.53.0-role.td:17]: Chatted offline. Yeah, in the current code we don't expect the mz_introspection role

**Comment by andrewrodriguez-m** [comment]
> I'm a little worried that changing the logged in username from `mz_introspection` to `mz_system` might cause some things to break, but I can't think of anything specific.

@MaterializeInc/cloud can you review this part? When someone logs in as mz_introspection we'll just pretend they said mz_system

**Comment by necaris** [comment]
The Prometheus-SQL exporter and a few CLI tools run as `mz_introspection`, but if it has the same permissions when pretending to be `mz_system` it _shouldn't_ break anything?

**Comment by andrewrodriguez-m** [comment]
> The Prometheus-SQL exporter and a few CLI tools run as `mz_introspection`, but if it has the same permissions when pretending to be `mz_system` it _shouldn't_ break anything?

Yeah. A lot of tests still use the mz_introspection so I think it should work

**Pull request merged**